src/Features/LanguageServer/Protocol/LanguageServerProtocol.cs

```diff
@@ -78,8 +80,8 @@ public Task<object> ExecuteWorkspaceCommandAsync(Solution solution, LSP.ExecuteC
         /// <param name="clientCapabilities">the client capabilities for the request.</param>
         /// <param name="cancellationToken">a cancellation token.</param>
         /// <returns>the location(s) of the implementations of the symbol.</returns>
-        public Task<object> FindImplementationsAsync(Solution solution, LSP.TextDocumentPositionParams request, LSP.ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
-            => ExecuteRequestAsync<LSP.TextDocumentPositionParams, object>(LSP.Methods.TextDocumentImplementationName, solution, request, clientCapabilities, cancellationToken);
+        public Task<LSP.SumType<LSP.Location, LSP.Location[]>?> FindImplementationsAsync(Solution solution, LSP.TextDocumentPositionParams request, LSP.ClientCapabilities clientCapabilities, CancellationToken cancellationToken)
```

Comment:
1. what is the value of this.  Why special case teh single-element result?  why not just put that in an an array of length 1?
1. Why use arrays instead of ImmutableArray?  Passing around mutable data seems unhappy.

Reply:
1.  The LSP originally only spec'd a single return for implementations.  They've since added other return types (array of Location)- https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_implementation .   Since it needs to be backwards compatible, the library introduced SumTypes to handle multiple possible return types.  If they had specified an array to begin with, we wouldn't need this.

2.  That's what is defined by the protocol and the VS library for it.  I would guess to be the simplest match to the protocol, but will let @tinaschrepfer comment.  It's probably too late to change as it would be a fairly large breaking change.
Since we don't use the result of this request, we can use immutable everywhere and just convert it back to array as soon as it leaves our control.